docs/manual/shared_memory.rst

```diff
@@ -1,54 +1,130 @@
+
+The memory of iceoryx is layed out as numbers of fixed-size segments, which taken together are iceoryx's memory pool. When a subscriber requests a block of memory from iceoryx, the smallest block which will fit the requested size and is available is provided to the subscriber from the pool.
+
+To ensure the best performance with the smallest footprint, the user is advised to configure iceoryx in such a manner that the memory pool only consists of blocks which will be useful to the exchanges to be done. Additionally, due to header information being sent along with the sample, the block size required from the pool is 64 bytes larger than the data type being exchanged. Lastly, iceoryx requires that the blocks be aligned to 4 bytes.
 
-Setup the environment before trying to run Cyclone DDS.
+Below follows an example of an iceoryx configuration file which has a memory pool of 2^15 blocks which can store data types of 16384 bytes (+ 64 byte header = 16448 byte block):
+
+.. code-block:: toml
+
+  [general]
+  version = 1
+
+  [[segment]]
+
+  [[segment.mempool]]
+  size = 16448
+  count = 32768
+
+The configuration file used is supplied to iceoryx using the -c parameter. Please save this file as *iox_config.toml* in your home directory for future use in the `Run`_ section.
+For detailed shared memory service documentation, the user is referred to `the iceoryx website <https://github.com/eclipse-iceoryx/iceoryx>`_.
+
+Cyclone DDS also needs to be configured correctly, to allow it to use shared memory exchange.
+
+The location where Cyclone DDS looks for the config file is set through the environment variable CYCLONEDDS_URI:
 
 .. code-block:: bash
 
-  cd ~/cyclone_iceoryx_ws
-  source install/local_setup.bash
+  export CYCLONEDDS_URI=file://cyclonedds.xml
+
+The following optional configuration parameters in SharedMemory govern how Cyclone DDS treats shared memory:
+
+* Enable
+
+  * when set to *true* enables cyclonedds to use shared memory for local data exchange
+
+  * defaults to *false*
+
+* SubQueueCapacity
 
-We need to enable shared memory configuration to run with iceoryx.
-Save the following xml as cyclonedds.xml.
+  * controls how many samples a reader using shared memory can hold before the least recent is discarded
+
+  * defaults to *256*
+
+* PubHistoryCapacity
+
+  * defines how many samples a shared memory writer will keep to send to late-joining subscribers
+
+  * defaults to *16*
+
+* SubHistoryRequest
+
+  * the number of samples a late-joining reader will request from a writer
+
+  * this can only be as many as were sent and at most PubHistoryCapacity
+
+  * defaults to *16*
+
+* LogLevel
+
+  * controls the output of the iceoryx runtime and can be set to, in order of decreasing output:
+
+    * *verbose*
+
+    * *debug*
+
+    * *info* (default)
+
+    * *warn*
+
+    * *error*
+
+    * *fatal*
+
+    * *off*
+
+Below follows an example of a Cyclone DDS configuration file supporting shared memory exchange:
 
```

Comment:
I wonder if we can get rid of these config options by incorporating them into the QoS.

The question then is what to do if the QoS request more than possible with iceoryx (fallback to max possible?, error? warning and do not use iceroyx?). Might be worth considering but for now this describes the settings as they are.

Reply:
I do not think the QoS settings are the correct place for configuration options in my opinion.

Reply:
Sure, it is fine as it reflects the current state, no changes needed (or requested). I was just wondering if the current state could be improved and what the consequences would be on this document.